packages/core/src/errors.ts

```diff
@@ -110,3 +121,35 @@ export class InvalidStringError<InnerError = unknown> extends ComposableError<In
     super(`Invalid string: "${expectation}"`, innerError);
   }
 }
+
+/**
+ * OgmiosCardanoNode and OgmiosTxSubmitProvider related errors
+ */
+export class UnknownCardanoNodeError<InnerError = unknown> extends ComposableError<InnerError> {
+  constructor(innerError: InnerError) {
+    super('Unknown CardanoNode error', innerError);
+  }
+}
+
+export class NotInitializedError extends CustomError {
+  constructor(methodName: string, moduleName: string) {
+    super(`${methodName} cannot be called until ${moduleName} is initialized`);
+  }
+}
+
+export const OgmiosClientErrors = {
```

Comment:
I think `core` package shouldn't export ogmios-specific types. For now it's ok to re-export types from `@cardano-ogmios/client` as we're already doing that for tx submission errors. I suggest renaming this back to `CardanoNodeErrors` and generalizing `WebSocketClosed` as it's probably too specific to Ogmios. How about `ConnectionError`?

Reply:
The idea behind moving the re-exposed Ogmios-related errors to commonplace and renaming it is that some of the errors are shared across [OgmiosCardanoNode](https://github.com/input-output-hk/cardano-js-sdk/pull/509/commits/092820216057f6489e8fc7c1136520075f0b1a99#diff-f39a864229b1800d755f14614cd8496fe47244069e7b95d5ae851156616cf34bR54-R83) and [OgmiosTxSubmitProvider](https://github.com/input-output-hk/cardano-js-sdk/pull/509/commits/092820216057f6489e8fc7c1136520075f0b1a99#diff-b7119a8f737f72567611a7b9e26aba55cab3472da2b8446ce73a6623a0b2ce93R63-R66), for instance `CardanoNodeNotInitializedError` -> `NotInitializedError`

Do you think we can keep the `CardanoNodeErrors` named export for all of them and use it in both Ogmios node and provider? @mkazlauskas

Reply:
Transactions are submitted to CardanoNode so yes, that makes sense. TxSubmissionError could be one of the variants of CardanoNodeError.

Reply:
Yeah, going to do it in a bit, about the TxSubmissionError I think it could be kept as it is in a separate file where we [compose and re-export](https://github.com/input-output-hk/cardano-js-sdk/blob/9d0f51fe4a3b8ae20c8e83b9209397cd99cc044b/packages/core/src/Cardano/types/TxSubmissionErrors.ts#L10-L84) bunch of TxSubmission errors under it

Reply:
Would be best to have just 1 export, could do a breaking refactor to use the new one under `CardanoNodeErrors.TxSubmissionError`

Reply:
The change is applied with [93bc87f](https://github.com/input-output-hk/cardano-js-sdk/pull/509/commits/93bc87f78b402cf27c397003b10d7676da5ea543), now the `TxSubmissionError` error is classified as `CardanoNodeError` and the usage is as follows `CardanoNodeErrors.TxSubmissionError`